src/components/documentComponents/documentLinkSearch/DocumentLinkList.js

```diff
@@ -0,0 +1,73 @@
+import { request } from "../../../utils.js";
+import DocumentLink from "./DocumentLink.js";
+
+export default function DocumentLinkList({ $parent, $target, initialState, onClose }) {
+  const $documentLinkList = document.createElement("ul");
+  $documentLinkList.className = "document-link-list";
+
+  $target.appendChild($documentLinkList);
+
+  this.state = initialState;
+
+  this.componentDidMount = async () => {
+    const documents = await request("");
+    const documentLinks = documents.map(({ id, title, documents }) => ({
+      id,
+      title,
+      documents: documents.map(({ id, title, documents }) => ({
+        id,
+        title,
+        documents: documents.map(({ id }) => ({ parentId: id })),
+      })),
+    }));
+    this.setState({ documentLinks });
+  };
+
+  this.componentDidMount();
+
+  this.setState = (nextState) => {
+    this.state = nextState;
+    this.render();
+  };
+
+  this.render = () => {
+    $documentLinkList.innerHTML = ` 
+    <p class="modal-title">페이지를 선택하세요</p>
+      <ul class="modal-list">
+      ${this.state.documentLinks
+        .map(
+          ({ id, title, documents }) => `
+          ${
+            title !== ""
+              ? `
+          <li data-id=${id} data-title="${title}" id="link" class="wrapper">
+            <i class="fa-regular fa-file"></i>
+            <span class="link-title">${title}</span>
+          </li>
+          `
+              : ""
+          }
+      `
+        )
+        .join("")}
+      </ul>`;
+  };
+
+  $documentLinkList.addEventListener("click", (event) => {
+    const { target } = event;
+    const { id, title } = target.closest(".wrapper").dataset;
```

Comment:
closest로 .wrapper를 찾았을때 요소가 존재하지 않는다면 종료하는 구문이 있다면 좋을 것 같습니다.
`$clickedElement` 이름은 대충 지은 이름이니 상황에 맞게 변경해서 사용하셔도 됩니다!

```suggestion
const $clickedElement = event.target.closest(".wrapper");
if (!$clickedElement) return;

const { id, title } = $clickedElement.dataset;
```

Reply:
방어코드가 중요하니 고려해봐야할 문제네요 감사합니다!